### DISCUSSION TITLE
[AI] A necessary step forward (or maybe lateral?)

### DISCUSSION THREAD

**Discussion by MarcoRavich** [SubtitleEdit/subtitleedit#8694] [Ideas]
Hi everyone,
1st of all a heartfelt/dutiful thanks to all devs/contributors of this project which I am a proud and satisfied user.

Given the increasing spread of so-called "artificial intelligence" (which at the moment is "just" machine learning, neural networks, large language models), I started to wonder what the future deserves for both fansubbers and softwares like this.

As IT tech - even if not a dev - I've always believed that technological evolution must be "ridden" (or, if you prefer, the "hare's race ahead" approach need to be adopted), so I began to look for what the open source scene offers in this field of usage.

There are of course many "voice to text" or "neural ocr" out there, but the step forward seems to be the "dubbing transpose" (aka translating+voice cloning+lipsyncing) IMHO.

I recently discovered some - even if I haven't yet personally tested any of - interesting open source projects which aims to perform such kind of "transposition":
- @jianchang512's [pyVideoTrans](https://github.com/jianchang512/pyvideotrans/blob/main/docs/EN/README_EN.md#video-translation-and-voiceover-tool);
- @JollyToday's [GhostCut](https://github.com/JollyToday/GhostCut-auto_video_translation#readme);
- @ThioJoe's [Auto Synced & Translated Dubs](https://github.com/ThioJoe/Auto-Synced-Translated-Dubs#readme)
- @FlorianEagox's [WeeaBlind](https://github.com/FlorianEagox/WeeaBlind#readme);
- @pranauv1's [AI-Video-Translation](https://github.com/pranauv1/AI-Video-Translation#readme);
- @R3gm's [SoniTranslate](https://github.com/R3gm/SoniTranslate#readme);

I would love to have an - both online and offline - "AutoDub" function in SE in the future...

...what do you think about ?

**Comment by niksedk** [comment]
SE already has a lot of AI - and has really had this for a long time.

Translate:
![image](https://github.com/user-attachments/assets/056cb500-8af4-4671-a44b-7613d51932f6)

Audio to text (besides Vosk):
![image](https://github.com/user-attachments/assets/f0835073-f0f2-4f73-94a6-cf39039d3567)

Text to speech:
![image](https://github.com/user-attachments/assets/db258a68-e827-4e17-a0ce-fca80538ec57)

OCR:
![image](https://github.com/user-attachments/assets/b6c741bf-e411-4510-a101-137d2e89bf26)

**Reply by MarcoRavich** [reply]
Thanks for the lightening-fast reply,
yes I know that SE uses many "AI" services for its features.

The question is if **audio** translation/transposition (= voice-isolation+voice-clone+text-to-speech+audio-merge) has been ever evaluated or it's just out of the project scope.

Thanks in advice.

**Reply by niksedk** [reply]
You can do this by first using "Audio to text" and then do "Text to speech", with an optional "Auto translate" between.

**Reply by MarcoRavich** [reply]
> You can do this by first using "Audio to text" and then do "Text to speech", with an optional "Auto translate" between.
>
...only voice-cloning is missing, then...

**Reply by cvrle77** [reply]
Service ElevenLabs can clone your voice.

**Reply by MarcoRavich** [reply]
> Service ElevenLabs can clone your voice.
>
Of course, but it's an online - closed source and paid - service.

Applio - which is opensource (MIT) - offers offline TTS/cloning (using your local computing power) instead.

Interesting "review" about:
[![FREE Text to Speech with YOUR Voice with Applio!](https://img.youtube.com/vi/gjggpadBgOo/maxresdefault.jpg)](https://youtu.be/gjggpadBgOo "Watch FREE Text to Speech with YOUR Voice with Applio! on YouTube")
